config/config_render.go

```diff
@@ -0,0 +1,320 @@
+			}
+		}
+		if v, ok := data[tag]; ok {
+			tmp := fmt.Sprintf("%v", v)
+			return w.Write([]byte(tmp))
+		}
+
+		v := composeVarKeyForTemplate(tag)
+		return w.Write([]byte(v))
+	})
+}
+
+// GetUnresolvedVars returns the vars in template that are no on data
+// In this case we don't use environment variables
+func (c *ConfigRender) GetUnresolvedVars(tpl *fasttemplate.Template,
+	data map[string]interface{}, useEnv bool) []string {
+	var unresolved []string
+	tpl.ExecuteFuncString(func(w io.Writer, tag string) (int, error) {
+		if useEnv {
+			if v, ok := c.findTagInEnvironment(tag); ok {
+				return w.Write([]byte(v))
+			}
+		}
+		if _, ok := data[tag]; !ok {
+			if !contains(unresolved, tag) {
+				unresolved = append(unresolved, tag)
+			}
+		}
+		return w.Write([]byte(""))
+	})
+	return unresolved
+}
+
+func contains(vars []string, search string) bool {
+	for _, v := range vars {
+		if v == search {
+			return true
+		}
+	}
+	return false
+}
+
+// GetVars returns the vars in template
+func (c *ConfigRender) GetVars(configData string) []string {
+	tpl, err := fasttemplate.NewTemplate(configData, startTag, endTag)
+	if err != nil {
+		return []string{}
+	}
+	vars := unresolvedVars(tpl, map[string]interface{}{})
+	return vars
+}
+
+func (c *ConfigRender) findTagInEnvironment(tag string) (string, bool) {
+	envTag := c.composeVarKeyForEnvirnonment(tag)
+	if v, ok := c.LookupEnvFunc(envTag); ok {
+		return v, true
+	}
+	return "", false
+}
+
+func (c *ConfigRender) composeVarKeyForEnvirnonment(key string) string {
+	return c.EnvinormentPrefix + "_" + strings.ReplaceAll(key, ".", "_")
+}
+
+func composeVarKeyForTemplate(key string) string {
+	return startTag + key + endTag
+}
+
+func readFileToString(filename string) (string, error) {
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+func unresolvedVars(tpl *fasttemplate.Template, data map[string]interface{}) []string {
+	var unresolved []string
+	tpl.ExecuteFuncString(func(w io.Writer, tag string) (int, error) {
+		if _, ok := data[tag]; !ok {
+			unresolved = append(unresolved, tag)
+		}
+		return w.Write([]byte(""))
+	})
+	return unresolved
+}
+
+func convertFileToToml(fileData string, fileType string) (string, error) {
+	switch strings.ToLower(fileType) {
+	case "json":
+		k := koanf.New(".")
+		err := k.Load(rawbytes.Provider([]byte(fileData)), json.Parser())
+		if err != nil {
+			return fileData, fmt.Errorf("error loading json file. Err: %w", err)
+		}
+		config := k.Raw()
+		tomlData, err := toml.Parser().Marshal(config)
+		if err != nil {
+			return fileData, fmt.Errorf("error converting json to toml. Err: %w", err)
+		}
+		return string(tomlData), nil
+	case "yml", "yaml", "ini":
+		return fileData, fmt.Errorf("cant convert from %s to TOML. Err: %w", fileType, ErrUnsupportedConfigFileType)
+	default:
+		log.Warnf("filetype %s unknown, assuming is a TOML file", fileType)
+		return fileData, nil
```

Comment:
Is this a safe assumption?

IMO, it would be better to check the filetype explicitly agains toml and in default return an error. WDYT?

```suggestion
	case "toml":
		return fileData, nil
	default:
		return fileData, fmt.Errorf("cant convert from %s to TOML. Err: %w", fileType, ErrUnsupportedConfigFileType)
```

Reply:
I fact, I did in that way because the unittest create temporary files that you can't decide the extension. To avoid renaming (and lost the feature of `CreateTemp`) I decide to assume that if you are passing a unknown extension try as TOML. If it's not is going to fail the parsing of the file